sdk/storage/AzureStorageBlob/Source/Data/BlockTransfer+CoreDataClass.swift

```diff
@@ -50,6 +51,7 @@ extension BlockTransfer {
         ) as? BlockTransfer else {
             fatalError("Unable to create BlockTransfer object.")
         }
+        transfer.clientRestorationId = clientRestorationId
```

Comment:
I kind of went back and forth on this. Block blob could easily read it from its parent. I'm not sure about multiblob. Maybe those don't need it at all because they delegate operations to their children?

Reply:
Right, that's what I'm thinking. The multiblob operation will really just be firing off a bunch of blob operations, with the "MultiBlob" grouping being notified of changes.

Reply:
Handled this by computing the property from the parent in BlockTransfer, and from the first child in MultiBlobTransfer. There is an edge case, of a MultiBlob transfer that has no children. In that case, I am just returning a bogus value that should never match an existing client, which will just cause the MultiBlobTransfer to transition to "failed". I don't love this approach, but it works. Can you think of a better alternative?

Reply:
Not really, but since MultiBlobTransfer is not yet implemented, it's probably fine. Realistically, it's nonsensical for a MultiBlobTransfer to contain no children. It must have at least one child (conceivably at least 2 or it's not really "multi").

Reply:
Agreed. Currently the `with` initializer doesn't require specifying any children. We can clean that up when we implement multiblob.